src/inmanta/data/__init__.py

```diff
@@ -775,21 +793,80 @@ def get_default_value(self) -> T:
 
     default_value = property(get_default_value)
 
-    def is_unique(self) -> bool:
-        return self._unique
-
-    unique = property(is_unique)
-
-    def is_reference(self) -> bool:
-        return self._reference
-
-    reference = property(is_reference)
+    @property
+    def ignore(self) -> bool:
+        return self._ignore
 
     def is_part_of_primary_key(self) -> bool:
         return self._part_of_primary_key
 
     part_of_primary_key = property(is_part_of_primary_key)
 
+    @property
+    def is_many(self) -> bool:
+        return self._is_many
+
+    def _validate_single(self, name: str, value: object) -> object:
+        """Validate a single value against the types in this field."""
+        if not (value.__class__ is self.field_type or isinstance(value, self.field_type)):
+            raise TypeError(
+                "Field %s should have the correct type (%s instead of %s)"
+                % (name, self.field_type.__name__, type(value).__name__)
+            )
+        return value
+
+    def validate(self, name: str, value: object) -> object:
+        """Validate the value against the constraint in this field. Treat value as list when is_many is true"""
+        if value is None and self.required:
+            raise TypeError("%s field is required" % name)
+
+        if value is None:
+            return None
+
+        if self.is_many:
+            if not isinstance(value, List):
+                TypeError("Field %s should be a list, but got %s" % (name, type(value).__name__))
+            else:
+                return [self._validate_single(name, v) for v in value]
+        return self._validate_single(name, value)
+
+    def from_db(self, name: str, value: object) -> object:
+        """Load values from database. Treat value as a list when is_many is true"""
+        if value is None and self.required:
+            raise TypeError("%s field is required" % name)
+
+        if value is None:
+            return None
+
+        if self.is_many:
+            if not isinstance(value, List):
+                TypeError("Field %s should be a list, but got %s" % (name, type(value).__name__))
+            else:
+                return [self._from_db_single(name, v) for v in value]
+        return self._from_db_single(name, value)
+
+    def _from_db_single(self, name: str, value: object) -> object:
+        """Load a single database value"""
```

Comment:
```suggestion
        """Load a single database value. Converts database representation to appropriately typed object."""
```
Wdyt? Same for `from_db`?

Reply:
good. updated